### PR TITLE
make starport more generic

### PR DESCRIPTION
**Pull request by waynenilsen**
Parameterize ETH and ETH: in the starport as immutable

### CODE REVIEW

File: ethereum/contracts/Starport.sol [hunk 1]
```diff
@@ -13,9 +13,9 @@ contract Starport {
     ICash immutable public cash;
 
     address immutable public admin;
-    address constant public ETH_ADDRESS = address(0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE);
-    bytes4 constant MAGIC_HEADER = "ETH:";
-    string constant ETH_CHAIN = "ETH";
+    address constant public NATIVE_TOKEN_ADDRESS = address(0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE);
```
Comment (hayesgm) [inline ethereum/contracts/Starport.sol]: This maybe should be immutable and different for different chains, e.g. `0xAaAAaaaaaAAAaaaAAaaaaaaAaAaaaAa` for mAtic?

File: ethereum/contracts/Starport.sol [hunk 2]
```diff
@@ -87,8 +89,8 @@ contract Starport {
      * @dev Use `lock` to lock CASH or collateral assets.
      */
     function lockEthTo(string memory chain, bytes32 recipient) public payable {
-        require(address(this).balance <= supplyCaps[ETH_ADDRESS], "Supply Cap Exceeded");
-        emit Lock(ETH_ADDRESS, msg.sender, chain, recipient, msg.value);
+        require(address(this).balance <= supplyCaps[NATIVE_TOKEN_ADDRESS], "Supply Cap Exceeded");
+        emit Lock(NATIVE_TOKEN_ADDRESS, msg.sender, chain, recipient, msg.value);
```
Comment (jflatow) [inline ethereum/contracts/Starport.sol]: Not sure its worth logging the `NATIVE_TOKEN_ADDRESS` every time?

**Comment by jflatow** [comment]
Looks mostly great, I think the issue with the test is that the constructor isn't being called with the new args which would set the `chainId`? It's funny that saddle is deploying without enough args, but that's what it looks like is happening

File: ethereum/contracts/Starport.sol (hunk 1 above)
Comment (waynenilsen) [reply ethereum/contracts/Starport.sol]: i think i'm ok with the native token address being the same across all evm chains - don't really see any reason why they should be different

File: ethereum/contracts/Starport.sol (hunk 2 above)
Comment (waynenilsen) [reply ethereum/contracts/Starport.sol]: yeah i think this just makes lock events generic

**Comment by waynenilsen** [comment]
> Looks mostly great, I think the issue with the test is that the constructor isn't being called with the new args which would set the chainId? It's funny that saddle is deploying without enough args, but that's what it looks like is happening

The issue was the encoding on the JS side as you pointed out in discord. The issue of the "hey, there aren't enough arguments" doesn't happen because `Starport` is not deployed directly it is deployed via `StarportHarness` which provides the additional CLI arguments

**Review by jflatow — APPROVED** [review]
Yay awesome 💪  So no interface change, and can be deployed for different native tokens. What are we planning to do about the governance interface (i.e. `executeProposal`)?

**Comment by waynenilsen** [comment]
pulled these changes into #330

this was a good place to review them though and get some consensus

**Pull request closed without merging**